packages/openneuro-app/src/scripts/refactor_2021/search/use-search-results.tsx

```diff
@@ -113,57 +149,103 @@ export const useSearchResults = () => {
     sortBy_selected,
   } = searchParams
 
-  const qStrings = []
-  if (keywords.length) qStrings.push(joinWithAND(keywords))
-  if (datasetType_selected) {
-  } // TODO: gql resolver level
-  if (datasetStatus_selected) {
-  } // TODO: gql resolver level
+  const boolQuery = new BoolQuery()
+  if (keywords.length)
+    boolQuery.addClause('must', simpleQueryString(sqsJoinWithAND(keywords)))
   if (modality_selected)
-    qStrings.push(`metadata.modalities: ${modality_selected}`)
+    boolQuery.addClause(
+      'filter',
+      matchQuery(
+        'latestSnapshot.summary.modalities',
+        modality_to_formats(modality_selected),
+      ),
+    )
   if (isActiveRange(ageRange))
-    qStrings.push(`metadata.ages: ${range(ageRange)}`)
-  if (isActiveRange(subjectCountRange)) {
-  } // TODO: https://discuss.elastic.co/t/painless-check-length-field-in-each-object-of-array/161699
+    boolQuery.addClause('filter', rangeQuery('metadata.ages', ...ageRange))
+  if (isActiveRange(subjectCountRange))
+    boolQuery.addClause(
+      'filter',
+      rangeListLengthQuery(
+        'latestSnapshot.summary.subjects.keyword',
+        subjectCountRange[0],
+        subjectCountRange[1],
+      ),
+    )
   if (diagnosis_selected)
-    qStrings.push(`metadata.dsStatus: ${diagnosis_selected}`)
+    boolQuery.addClause(
+      'filter',
+      matchQuery('metadata.dxStatus', diagnosis_selected),
+    )
   if (tasks.length)
-    qStrings.push(`latestSnapshot.summary.tasks: ${joinWithAND(tasks)}`)
+    boolQuery.addClause(
+      'must',
+      simpleQueryString(sqsJoinWithAND(tasks), [
+        'latestSnapshot.summary.tasks',
+      ]),
+    )
   if (authors.length)
-    qStrings.push(`metadata.seniorAuthor: ${joinWithAND(authors)}`)
-  if (gender_selected !== 'All') {
-    qStrings.push(
-      `latestSnapshot.summary.subjectMetadata.sex: ${gender_selected}`,
+    boolQuery.addClause(
+      'must',
+      matchQuery('metadata.seniorAuthor', joinWithOR(authors)),
     )
+  if (gender_selected !== 'All')
+    boolQuery.addClause(
+      'filter',
+      matchQuery('latestSnapshot.summary.subjectMetadata.sex', gender_selected),
+    )
+  if (date_selected !== 'All Time') {
+    const d =
+      date_selected === 'Last 30 days'
+        ? 30
+        : date_selected === 'Last 180 days'
+        ? 180
+        : 365 // 'Last 365 days'
```

Comment:
This is pretty confusing, how about refactoring it to the if else like it was before?